### ISSUE TITLE
Design PCB for sensor modules

### ISSUE THREAD

**Issue by jeffrobots** [ECESeniorDesign/greenhouse-sensors#8]
KiCad will be used to develop sensor module PCB files as soon as the prototype circuits are proven. The packages will use the smallest sensor packages possible. They should also include necessary pull-up resistors and bypass capacitors. The datasheets contain layout information (as well as connection information) and should be referenced often. 

Required:
- [x] Proof the sensor module prototype in its entirety. 
- [x] Move design to schematic and PCB Gerber file
- [x] Implement status LEDs and analog sensor power control
- [x] Design review of PCB layout

Recommended:
- [x] Move sensor prototype to a soldered perfboard and proof again
- [x] Order and test received PCBs

**Comment by jeffrobots** [comment]
@cconklin In case you're interested, ![I've made a rendering of the sensor module](https://github.com/ECESeniorDesign/greenhouse-sensors/blob/master/hardware/sensor_module_0_1.jpg). 

Even if you're not interested, I think I should go over the schematics with someone in the department before I send them off for fabrication. It would probably also be a good idea if you at least checked the schematics and we did a design review. That's a little ways away though. I still need to get some new components in just to verify everything. I also need to figure out if any sort of interface/amplification will be needed on our pH and soil moisture meters.

**Comment by cconklin** [comment]
Agreed. We can discuss it further with Derefinko on Monday.

**Comment by jeffrobots** [comment]
The last required tasks before the PCB can be ordered are verifying the BJT soil sensor power control as well as looking into adding status LEDs where appropriate.

**Comment by jeffrobots** [comment]
As of 6536d1a24c97237ee98df545fb8141bdaf6287d7 I have confirmed that using a transistor to power the soil sensor on and off will work. 

I will be reworking the PCB files tomorrow to include this and a status LED.

**Comment by jeffrobots** [comment]
@cconklin I just uploaded the gerbers to OSHPark. I'll figure out how to order through the department on monday. 

If you want to see it, take a look here. 

<a href="https://oshpark.com/shared_projects/Xo4KOmY4"><img src="https://a800d827b6de8403a51e-6ffc2e718631809086ea40332b2055f7.ssl.cf1.rackcdn.com/assets/badge-5b7ec47045b78aef6eb9d83b3bac6b1920de805e9a0c227658eac6e19a045b9c.png" alt="Order from OSH Park"></img></a>

**Comment by jeffrobots** [comment]
@cconklin 

Everything works perfectly on the board except for, as you might be able to guess, the light sensor. 

The status LED isn't working, and I thought it was an issue of the resistor size. After closer inspection, it seems possible that I actually wired it up backwards. They don't really have very visible markings for determining which way to put it and I wasn't really thinking straight when I was soldering it. 

Are you free at all tomorrow afternoon in case I need a second pair of eyes on the light sensor? I've looked it over a bit and honestly have no idea what its problem is. Also, you might want to take a look at how amazing it is before spring break.

**Comment by cconklin** [comment]
I probably won't be available.

**Comment by jeffrobots** [comment]
After testing continuity on the pads of the light sensor, I discovered that the SMBus pads on the light sensor are indeed swapped. Luckily, @cconklin convinced me to put jumper pads so that I can switch them around if need be. I grabbed a few jumpers from the lab and will stick them on the board and cut the traces as soon as possible.

The error comes from the fact that the PCB footprint does not use the same numbering scheme as the schematic model. On the schematic, the SMBData pad is at pin 5, which is at the top. The footprint places pin 5 at the bottom. This resulted in SMBData and SMBClk getting swapped around in the transition from schematic to PCB.

I'll make a new issue for revision of the PCB files.
